### ISSUE TITLE
WrongDamage

### ISSUE THREAD

**Issue by e7a-0-e7a** [PluginBugs/Issues-ItemsAdder#1926]
### Discord tag (optional)

Rhaven#1906

### What happened?

I don't do any real damage.
Example: Diamond sword + enemy = 1 - 3 damage.
Diamond sword + Enchantment Sharpness lvl 500 = 16 damage

I then only did Itemsadder + Lonelib on my test server without any other plugins and the problem persists.

If I take ItemsAdder away then I do the normal damage again.

### Steps to reproduce the issue

1. put Diamond Sword (or any other) in your Hand
2. Hit other Player or Mob

### Server version

paper-1.18.2-379

### ItemsAdder Version

ItemsAdder Version: 3.2.3f

### ProtocolLib Version

ProtocolLib Version: 5.0.0-SNAPSHOT-b597

### LoneLibs Version

LoneLibs Version 1.0.21

### ItemsAdder config.yml

```shell
metrics: true
update-checker: true
disable-plugin-loading-messages_DONT_ASK_ME_FOR_SUPPORT_IF_YOU_SET_TRUE__THANKS: false
resource-pack:
  hosting:
    no-host:
      enabled: false
    auto-external-host:
      enabled: false
    self-host:
      enabled: false
      server-ip: 127.0.0.1
      pack-port: 8163
    external-host:
      enabled: true
      url: https://www.dropbox.com/s/rrbagb5r30sc6e0/pack.zip?dl=0
      skip-url-file-type-check___DONT_ASK_HELP_IF_SET_TRUE: true
  apply-on-join: true
  kick-player-on-decline: false
  kick-player-on-fail: false
  delay-ticks: 1
  notify-other-plugins-interfering: false
  title:
    enabled: true
  protect-player:
    black-screen: false
    hide-hud: false
    cancel-movement: false
    lock-player: false
  command:
    usage-cooldown-seconds: 60
  custom-font:
    enabled: false
    name: ipaexg.ttf
    shift: '[0,0]'
    size: 11.0
    oversample: 2.0
  thin-font:
    enabled: false
  zip:
    protect-file-from-unzip:
      enabled: true
      extreme: true
    compress-png-files: true
    compress-json-files: true
    overwrite-vanilla-models: false
    overwrite-vanilla-lang-json: false
    generate-custom-armors-textures:
      vanilla_1_17: true
      optifine: true
    disable-overwrite-leather-armor-layers-textures: false
    fix_transparent_noteblocks__disable-for-more-performance: false
    ignore-files-ext:
    - db
    - yml
    - xcf
    - psd
    - html
    - gz
    - rar
    - zip
    - 7z
    - bbmodel
    - iaentitymodel
  items:
    CustomModelData-starting-value:
      ALL: 10000
      DIAMOND: 11000
config_files:
  lang: en
  dictionaries-lang: en
  extract-default-configs: true
  legacy_pack: false
blocks:
  light-api: true
  fix-liquids-flow: true
  fix-floating-blocks: true
  fix-glitched-blocks:
    enabled: true
    only-new-chunks: false
    search-also-in-these-biomes:
      enabled: false
      biomes:
      - BEACH
      - BIRCH_FOREST
      - BIRCH_FOREST_HILLS
      - FOREST
      - MOUNTAINS
      - MOUNTAIN_EDGE
      - GRAVELLY_MOUNTAINS
      - PLAINS
      - RIVER
  cancel-break-wrong-tool:
    enabled: true
    send-actionbar: true
  campfire:
    unlit-on-place: true
    allow-any-item: true
  disable-REAL: false
  disable-REAL_TRANSPARENT: false
  disable-REAL_NOTE: false
  disable-REAL_WIRE: true
  disable-FIRE: false
items:
  fix-show-attack-and-damage-as-lore: true
  give_death_location_maps: true
  hide-custom-durability-lore: false
liquids:
  enabled: true
huds:
  enabled: true
loots:
  allow-loots-drop-from-spawners-using-silk-touch: false
mobs:
  disable-MOB-behaviour__DISABLE_AT_YOUR_OWN_RISK: false
font_images:
  replace-in-customitems-name-and-lore: true
  chat:
    enabled: true
    doublecheck-permission__disable-for-more-performance: false
    add-hover-suggestion: true
  command:
    enabled: true
    commandblocks: true
    excluded:
    - home
    - sethome
    - tpa
  sign:
    enabled: true
  book:
    enabled: true
  anvil:
    enabled: true
  inventory-title:
    enabled: true
  vault-prefix-suffix:
    enabled: true
    update-every-ticks: 1
  scoreboard-teams:
    enabled: false
    update-every-ticks: 3600
  player-display-name:
    enabled: true
    update-every-ticks: 1
effects:
  hide-scoreboard-numbers: false
  text-effects:
    enabled: true
    replace-in-customitems-name-and-lore: true
    chat:
      enabled: true
    sign:
      enabled: true
    book:
      enabled: true
    anvil:
      enabled: true
entities:
  max-furniture-vehicles-per-chunk: 30
  allow-removing-old-furnitures_2_3_11: false
  custom-entitites:
    enabled: true
    emotes: true
    interval-rendering-packets-ticks: 1
    optimized-packets: true
recipes:
  hide-vanilla-unrecognized-recipe-errors: true
  fix-packet-too-large-kick: false
  custom-recipes:
    unlock:
      all-on-join: false
      on-pickup:
        any: false
        ingredients: true
  remove-vanilla-recipes:
    enabled: true
    list:
      DIAMOND_SWORD: false
      IRON_INGOT: false
debug:
  items:
    log-files-loading: false
    log-items-loading: false
    log-items-loading-disabled: false
    log-font-images-loading: false
    log-huds-loading: false
    log-attribute-modifiers-loading: false
    log-warnings: false
    log-loots-loading: false
    log-loots-loading-disabled: false
    log-armors-loading: false
  recipes:
    log-recipes-loading: false
    log-recipes-loading-disabled: false
    log-recipes-removed-vanilla: false
    log-recipes-unknown-item: false
    log-fix-packet-too-large: false
  resource-pack:
    log-extract-default-resources: true
    log-not-extract-default-resources-already-exist: true
  blocks:
    log-save: false
    log-save-on-unload: false
    log-mined-unknown-blocks: true
  worlds-populators:
    log-loading: false
    log-loading-disabled: false
    log-loading-no-populators: false
    log-loading-no-world: true
    log-generation: false
  trees-populators:
    log-loading: false
    log-loading-disabled: false
    log-loading-no-populators: false
    log-loading-no-world: true
    log-generation: false
  surface-decorators:
    log-loading: false
    log-loading-disabled: false
    log-loading-no-decorators: false
    log-loading-no-world: true
    log-generation: false
  cave-decorators:
    log-loading: false
    log-loading-disabled: false
    log-loading-no-decorators: false
    log-loading-no-world: true
    log-generation: false
  player-stats:
    log-save: false
  performance:
    loading-ms: true
  huds:
    log-triggers: false
  other:
    log-dictionaries-loading: false
    log-liquids-loading: false
    log-minecraft_lang_overwrite-loading: false
  categories:
    log-total-loaded: true
    log-single-loaded: false
    log-single-loaded-overwrite: false
    log-single-skip: false
    log-icon-failed-load: false
  entities:
    log-files-loading: false
    log-entities-loading: false
    log-entities-loading-disabled: false
chat-preview: false
```

### Screenshots/Videos (you can drag and drop files or paste links)

Without ItemsAdder:
https://gyazo.com/e62c7fad34e73c4e032fbdec85630800

With ItemsAdder:
https://gyazo.com/dbeb41ef097317f95b41f333c7931621
https://gyazo.com/5ac8c6ffd42e14d5ad178a604f616b1b

Full LOG:
https://hastebin.rhaven.de/ociniwoyom.bash

**Comment by LoneDev6** [comment]
I cannot reproduce the issue.

https://user-images.githubusercontent.com/27242001/186899670-ae01309c-7e1c-48f0-a627-e33f0e800b3d.mp4

**Comment by e7a-0-e7a** [comment]
And what could it be?

Because even if I have only ItemsAdder + Lonelibs on my server, I have this problem. And I only do normal damage when ItemsAdder is no longer on the server.

If only 1 plugin is on it (ItemsAdder+Lonelib) otherwise no more. And I can not do the right damage, then it can only be the plugin. Especially since everything works normally again when this plugin is no longer on it.

Or is there something else that can cause this problem, independent of the plugin, which I did not know yet?

And on the Discord have meanwhile others with the same problem announced.

**Comment by LoneDev6** [comment]
It wasn't necessary to open a PayPal dispute...
Do you think that opening a refund request would speedup my job?
I'd like to help you but I need more info in order to identify the issue.
Please consider closing the issue report, I will do my best to identify and fix the issue.

I probably have misunderstood the issue report.


Let me rephrase your issue report and then please confirm that it's what you meant:

> VANILLA Diamond swords make 7-10 damage when ItemsAdder is installed.
They should not deal this kind of damage but only 3-4.


Is this correct?

**Comment by e7a-0-e7a** [comment]
I do not have hope that it will be faster. I was just reacting to your answer which was only "I can't reproduce it".

Likewise, this was also a reaction to your answer at Spigotmc, because that a diamond sword makes max 3 damage is simply wrong.

I have now waited almost 1 week for a solution and the only thing I got as an answer was "I can not reproduce it". These were my reasons why I have switched on paypal. Because these answers gave me to understand that no solution will follow.

The problem is that WITH ItemsAdder 1 - max 3 damage is done to player and mob (with a diamond sword) THAT is the wrong damage.
With enchantment Sharpness 250 I do max 17 damage. Also THAT is the wrong damage. And all WITH ItemsAdder.

WITHOUT ItemsAdder I get the normal damage of 5 - 7 damage to player and mob without enchantment.


[WITH ItemsAdder](https://gyazo.com/c7b23bf5199a522c0abfab6a2f094df4)

[WITHOUT ItemsAdder](https://gyazo.com/2ffffc0abd7bde35778c66eb090f6671)

I would be happy if we get this solved, because I would like to use the plugin.

**Comment by LoneDev6** [comment]
I literally made a video to show you what happens on a clean server with only ItemsAdder:
Diamond Sword deals 7 hearts damage.

I installed DamageIndicator, the plugin you're using.
This is the damage shown in the Minecraft wiki too:
https://minecraft.fandom.com/wiki/Sword

![image](https://user-images.githubusercontent.com/27242001/187024832-e06c666e-baf7-4600-8e51-2e27382a98be.png)

## Detailed test
Here in this test I attack a pig with a vanilla Diamond Sword and ItemsAdder is installed.
As you can see it deals the correct damage.
I also spam-attack to show that it deals a lower damage (1.8), because of the vanilla attack cooldown, which works fine.

Then I disable ItemsAdder, the damage is the same, correct one.

https://user-images.githubusercontent.com/27242001/187024882-b3b0a0b6-bb5e-4912-b71f-bd3bb31a0412.mp4

There is another plugin on your server which is interfering.
Please remove EVERY other plugin and test.

When I tell you "I cannot reproduce the issue" is not because I want to shut your mouth and run away with the money, I am just trying to find the cause of the issue.
I hope you understand how things are managed in software development, I cannot simply discover the source of an issue if I cannot reproduce the problem on my server.

**Comment by e7a-0-e7a** [comment]
> There is another plugin on your server which is interfering.
> Please remove EVERY other plugin and test.

I wrote in the bug report and in one of my answers, here, that I tested it WITHOUT any other plugins on my server.

I have also first thought that it is due to another plugin have ALL plugins gradually turned off. And then finally ONLY Itemsadder pulled to another server and tried it there again. Even there, on the other server, WITHOUT other plugins it does not work.

I am aware that if you can't reproduce it, it is hard to find the error. Or to explain what the problem is. If it doesn't work when I don't have any other plugins on it, except ItemsAdder, then it can only be that.

Maybe it's the Minecraft version, I wrote I use the 1.18.2 and you try to reproduce it with the 1.19. I do not know. I only know that ItemsAdder is currently unusable for me because it causes this error.

**Comment by LoneDev6** [comment]
Are we actually sure that the damage is wrong and it's not a bug of DamageIndicator?

**Comment by e7a-0-e7a** [comment]
So I am sure that the damage is wrong and not only wrong displayed.

In the video you can see that I have made 2x /pl. 1x to show that the damage indicator works because only this was on the server.

and 2x to show that I have only Itemsadder + dependencies on it and the damage indicator.

in addition you can see that I hit without ItemsAdder 2x and the pig dies and I have to hit with ItemsAdder several times.

[Proof is the Damage, not the Plugin Damage Indicator (Video)](https://gyazo.com/8e18c8320e3e4e7c4b8528c11720374b)

**Comment by LoneDev6** [comment]
I hope you can change your mind about your review, I did my best but can't find the cause of this particular issue.
Thanks a lot

![image](https://user-images.githubusercontent.com/27242001/187214216-0d841c03-7641-4622-aef2-e613c48fd59e.png)

**Comment by e7a-0-e7a** [comment]
That is a pity. But thanks for that you have searched the error

**Comment by LoneDev6** [comment]
I re-added you to the buyers list for free so that you can continue using the plugin even without owing it.
I did that in case you ever find the time and motivation you can try to hunt down the issue source and let me know.

Have a nice week

**Comment by e7a-0-e7a** [comment]
Thank you very much! As soon as I find something, I will definitely let you know.
